src/components/modal/tenants/AddEditTenantDialog.tsx

```diff
@@ -50,72 +59,109 @@ export default class AddTenantManuallyDialog extends React.Component<Props, Stat
 
   public constructor(props: Props) {
     super(props);
-    let tenantEndpointClouds: EndpointCloud[];
-    if (__DEV__) {
-      tenantEndpointClouds = Configuration.DEVELOPMENT_ENDPOINT_CLOUDS;
-    } else {
-      tenantEndpointClouds = Configuration.ENDPOINT_CLOUDS;
-    }
+    const tenantEndpointClouds: EndpointCloud[] = Configuration.getEndpoints();
+    this.tenant = props.tenants?.[props.tenantIndex];
     // Set initial state
     this.state = {
       newTenantSubDomain: null,
       newTenantName: null,
       tenantEndpointClouds,
-      newTenantEndpointCloud: Configuration.ENDPOINT_CLOUDS?.[0],
-      showSubdomainAlreadyUsedError: false
+      newTenantEndpointCloud: null,
+      errorSubdomainAlreadyUsed: false
     };
   }
 
+  public componentDidMount() {
+    const { mode } = this.props;
+    if (mode === TenantDialogMode.EDIT) {
+      const newTenantEndpointCloud = Configuration.getEndpoints()?.find((e) => e.endpoint === this.tenant.endpoint);
+      this.setState({
+        newTenantSubDomain: this.tenant?.subdomain,
+        newTenantName: this.tenant?.name,
+        newTenantEndpointCloud
+      });
+    }
+  }
+
   public render() {
     const style = computeStyleSheet();
     const modalCommonStyle = computeModalCommonStyle();
     const { newTenantSubDomain, newTenantName, newTenantEndpointCloud } = this.state;
-    const { back } = this.props;
-
+    const { back, mode, withCancel } = this.props;
+    const addButtons = [
```

Comment:
[low] You could have put this code in the template for consistency with other screens ;-)

Reply:
I don't manage to make the linter happy in the template

Reply:
We can discuss if this is what we want and if yes, change the linter rules

Reply:
I think that it would be best in a sepearate render method, otherwise it makes the code harder to read